simtools/simulator.py

```diff
@@ -0,0 +1,740 @@
+        'dataDirectory': '(..)/data',
+        'primary': 'gamma',
+        'zenith': 20 * u.deg,
+        'azimuth': 0 * u.deg,
+        'viewcone': 0 * u.deg,
+        # ArrayModel
+        'site': 'North',
+        'layoutName': '1LST',
+        'modelVersion': 'Prod5',
+        'default': {
+            'LST': '1'
+        },
+        'M-01': 'FlashCam-D'
+    }
+
+    Attributes
+    ----------
+    label: str
+        Instance label.
+    config: NamedTuple
+        Configurable parameters.
+    arrayModel: ArrayModel
+        Instance of ArrayModel.
+
+    Methods
+    -------
+    run(inputFileList):
+        Run simulation.
+    submit(inputFileList, submitCommand=None, extraCommands=None, test=False):
+        Submit a run script as a job. The submit command can be given by submitCommand \
+        or it will be taken from the config.yml file.
+    printHistograms():
+        Print histograms and save a pdf file.
+    getListOfOutputFiles():
+        Get list of output files.
+    getListOfInputFiles():
+        Get list of input files.
+    getListOfLogFiles():
+        Get list of log files.
+    printListOfOutputFiles():
+        Print list of output files.
+    printListOfInputFiles():
+        Print list of output files.
+    printListOfLogFiles():
+        Print list of log files.
+    """
+
+    def __init__(
+        self,
+        label=None,
+        simulator=None,
+        filesLocation=None,
+        simtelSourcePath=None,
+        configData=None,
+        configFile=None,
+    ):
+        """
+        Simulator init.
+
+        Parameters
+        ----------
+        label: str
+            Instance label.
+        simulator: str
+            Type of simulator called (implemented is simtel and corsika).
+        filesLocation: str or Path.
+            Location of the output files. If not given, it will be set from \
+            the config.yml file.
+        simtelSourcePath: str or Path
+            Location of source of the sim_telarray/CORSIKA package.
+        configData: dict
+            Dict with shower or array model configuration data.
+        configFile: str or Path
+            Path to yaml file containing configurable data.
+        """
+        self._logger = logging.getLogger(__name__)
+        self._logger.debug("Init Simulator {}".format(simulator))
+
+        self.label = label
+        self._set_simulator(simulator)
+        self.runs = list()
+        self._results = defaultdict(list)
+
+        self._set_file_locations(filesLocation, simtelSourcePath)
+        self._loadConfigData(configData, configFile)
+
+        self._setSimulationRunner()
+
+    def _set_simulator(self, simulator):
+        """
+        Set and test simulator type
+
+        Parameters
+        ----------
+        simulator: str
+            ype of simulator called (implemented is simtel and corsika).
+
+        Raises
+        ------
+        gen.InvalidConfigData
+
+        """
+
+        self.simulator = simulator.lower()
```

Comment:
Shouldn't the order of these two lines be reversed? I.e., do not set the simulator before checking its validity?

Reply:
Agree